src/backends/sqlite3/session.cpp

```diff
@@ -162,10 +180,58 @@ void sqlite3_session_backend::rollback()
     execude_hardcoded(conn_, "ROLLBACK", "Cannot rollback transaction.");
 }
 
+struct SeqCtxt
+{
+    long long value_;
+    bool filledIn_;
+};
+
+static int GetOneLong(void* ctxt, int valueNum, char** values, char**)
+{
+    SeqCtxt* seqCtxt = (SeqCtxt*)ctxt;
+    seqCtxt->value_ = 0;
+    seqCtxt->filledIn_ = true;
+    char* ptr = NULL;
+    if (valueNum == 1 && values[0])
+        seqCtxt->value_ = strtol(values[0], &ptr, 10);
+    return 0;
+}
+
 bool sqlite3_session_backend::get_last_insert_id(
-    session & /* s */, std::string const & /* table */, long long & value)
+    session &, std::string const & table, long long & value)
 {
-    value = static_cast<long long>(sqlite3_last_insert_rowid(conn_));
+    char *zErrMsg = NULL;
+    SeqCtxt seqCtxt;
+    seqCtxt.filledIn_ = false;
+    seqCtxt.value_ = 0;
+    if (sequenceTableExists_ || SequenceTableExists(conn_))
+    {
+        // Once the sqlite_sequence table is created (because of a column marked AUTOINCREMENT)
+        // it can never be dropped, so don't search for it again.
+        sequenceTableExists_ = true;
+
+        std::string const query = "select seq from sqlite_sequence where name ='" + table + "'";
```

Comment:
I'm a bit nervous with using table name as part of the query. It's probably unlikely that it wouldn't be under program control, but if it can ever come from an untrusted source, we've got an SQL injection here.